Apps/frontend/src/views/Layout.svelte

```diff
@@ -0,0 +1,114 @@
+<script>
+  import { onDestroy, onMount } from "svelte";
+  import Logo from "./Logo.svelte";
+  import Indicators from "./Indicators.svelte";
+  import ExpandableControlPanel from "./ExpandableControlPanel.svelte";
+  import ControlPanelBottom from "./ControlPanelBottom.svelte";
+  import GeneralButtons from "./GeneralButtons.svelte";
+  import StartStopButton from "../components/StartStopButton.svelte";
+  import Waves from "../components/Waves.svelte";
+  import CoordinateSystem from "./CoordinateSystem.svelte";
+  import { NUM_CHANNELS } from "../const";
+  import { osciEnabled } from "../stores";
+  import { logSocketCloseCode } from "../helper";
+
+  $: innerWidth = 0;
+  $: innerHeight = 0;
+  $: controlPanelBottomHeight = 0;
+
+  let scalesY = Array(NUM_CHANNELS).fill(1); // 1V per horizontal line
+
+  let waveElement;
+  let indicatorElement;
+
+  let socket;
+
+  // ----- Svelte lifecycle hooks -----
+  onMount(() => {
+    socket = createSocket();
+
+    socket.onmessage = socketOnMessage;
+    socket.onclose = socketOnClose;
+  });
+
+  onDestroy(() => {
+    socket.close();
+  });
+
+  // -----business logic functions -----
+  const createSocket = () => {
+    console.log("Connecting to :" + import.meta.env.VITE_BACKEND_WEBSOCKET);
+    let socket = new WebSocket(import.meta.env.VITE_BACKEND_WEBSOCKET);
+    socket.binaryType = "arraybuffer";
+
+    return socket;
+  };
+
+  /**
+   * On receiving a socket message, update buffer of waves.
+   *
+   * @param {MessageEvent} messageEvent - has poperty (Float64Array) data
+   */
+  const socketOnMessage = (messageEvent) => {
```

Comment:
Feels wrong to have the websocket and update buffer logic in a file named Layout. Not a big deal as it wasn't much better before. (originally Oscilloscope.svelte)

Reply:
Git didn't pick up on it cause it wasn't ;)
I created `Layout.svelte` separately to build the whole Layout from scratch so I don't have to mingle with the existing one. Will rename to `Oscilloscope.svelte`, was a good point.